acapy_agent/anoncreds/holder.py

```diff
@@ -231,25 +230,17 @@ async def _finish_store_credential(
         rev_reg_def: Optional[dict] = None,
     ) -> str:
         credential_data = cred_recvd.to_dict()
-        schema_id = cred_recvd.schema_id
-        schema_id_parts = re.match(r"^(\w+):2:([^:]+):([^:]+)$", schema_id)
-        if not schema_id_parts:
-            raise AnonCredsHolderError(f"Error parsing credential schema ID: {schema_id}")
-        cred_def_id = cred_recvd.cred_def_id
-        cdef_id_parts = re.match(r"^(\w+):3:CL:([^:]+):([^:]+)$", cred_def_id)
-        if not cdef_id_parts:
-            raise AnonCredsHolderError(
-                f"Error parsing credential definition ID: {cred_def_id}"
-            )
+        registry = self.profile.inject(AnonCredsRegistry)
+        schema_info = await registry.get_schema_info_by_id(credential_data["schema_id"])
 
         credential_id = credential_id or str(uuid4())
         tags = {
-            "schema_id": schema_id,
-            "schema_issuer_did": schema_id_parts[1],
-            "schema_name": schema_id_parts[2],
-            "schema_version": schema_id_parts[3],
-            "issuer_did": cdef_id_parts[1],
-            "cred_def_id": cred_def_id,
+            "schema_id": credential_data["schema_id"],
+            "schema_issuer_id": schema_info.issuer_id,
+            "schema_name": schema_info.name,
+            "schema_version": schema_info.version,
+            "issuer_id": credential_definition["issuerId"],
+            "cred_def_id": cred_recvd.cred_def_id,
```

Comment:
Is the `schema_issuer_did` to `schema_issuer_id` and `issuer_did` to `issuer_id` going to cause issues in wallets with existing credential records? Do we need an upgrade script for this?

Reply:
I reverted this change. I don't want to mess up the holder tags or require a migration for this.